dozer/cogs/moderation.py

```diff
@@ -292,88 +298,133 @@ async def on_message(self, message):
 
+    @Cog.listener()
+    async def on_raw_message_delete(self, payload):
+        """When a message is deleted and its not in the bot cache, log it anyway."""
+        if payload.cached_message:
+            return
+        guild = self.bot.get_guild(int(payload.guild_id))
+        message_channel = self.bot.get_channel(int(payload.channel_id))
+        message_id = int(payload.message_id)
+        message_created = discord.Object(message_id).created_at
+        embed = discord.Embed(title="Message Deleted",
+                              description="Message Deleted In: " + str(message_channel.mention),
+                              color=0xFF00F0, timestamp=message_created)
+        embed.add_field(name="Message", value="N/A", inline=False)
+        embed.set_footer(text=f"MessageID: {str(message_id)}; Sent at")
+        message_log_channel = await self.edit_delete_config.query_one(guild_id=guild.id)
+        if message_log_channel is not None:
+            channel = guild.get_channel(message_log_channel.messagelog_channel)
+            if channel is not None:
+                await channel.send(embed=embed)
+
     @Cog.listener('on_message_delete')
     async def on_message_delete(self, message):
         """When a message is deleted, log it."""
         if message.author == self.bot.user:
             return
-        e = discord.Embed(type='rich')
-        e.title = 'Message Deletion'
-        e.color = 0xFF0000
-        e.timestamp = datetime.datetime.utcnow()
-        e.add_field(name='Author', value=message.author)
-        e.add_field(name='Author pingable', value=message.author.mention)
-        e.add_field(name='Channel', value=message.channel)
-        if 1024 > len(message.content) > 0:
-            e.add_field(name="Deleted message", value=message.content)
-        elif len(message.content) != 0:
-            e.add_field(name="Deleted message", value=message.content[0:1023])
-            e.add_field(name="Deleted message continued", value=message.content[1024:2000])
-        elif len(message.content) == 0:
-            for i in message.embeds:
-                e.add_field(name="Title", value=i.title)
-                e.add_field(name="Description", value=i.description)
-                e.add_field(name="Timestamp", value=i.timestamp)
-                for x in i.fields:
-                    e.add_field(name=x.name, value=x.value)
-                e.add_field(name="Footer", value=i.footer)
+        audit = await self.check_audit(message.guild)
+        embed = discord.Embed(title="Message Deleted",
+                              description=f"Message Deleted In: {str(message.channel.mention)}\nSent by: {str(message.author.mention)}",
+                              color=0xFF0000, timestamp=message.created_at)
+        embed.set_thumbnail(url=message.author.avatar_url)
+        if audit:
+            if str(audit.target) == str(message.author.name + "#" + message.author.discriminator):
```

Comment:
What's the reasoning for comparing equality by `str(...)`? Does `audit.target == message.author` not do the expected thing here?